app/src/semaphore.h

```diff
@@ -61,24 +64,31 @@ class Semaphore {
     snprintf(buffer, kBufferSize, "%016llx",
              static_cast<unsigned long long>(  // NOLINT
                  reinterpret_cast<intptr_t>(this)));
-
 #if FIREBASE_PLATFORM_OSX
-    // Sandbox mode is currently only available for macOS.
+    // Append custom semaphore names, if present, to support macOS Sandbox
+    // mode.
     std::string semaphore_name = util::GetSandboxModeSemaphorePrefix();
     if (semaphore_name.empty()) {
-      semaphore_name = "/firebase-";
+      semaphore_name = FIREBASE_SEMAPHORE_DEFAULT_PREFIX;
     }
 #else
-    std::string semaphore_name = "/firebase-";
+    std::string semaphore_name = FIREBASE_SEMAPHORE_DEFAULT_PREFIX;
 #endif  // FIREBASE_PLATFORM_OSX
 
     semaphore_name.append(buffer);
-
     semaphore_ = sem_open(semaphore_name.c_str(),
                           O_CREAT | O_EXCL,   // Create if it doesn't exist.
                           S_IRUSR | S_IWUSR,  // Only the owner can read/write.
                           initial_count);
 
+    // Check for errors.
+#if FIREBASE_PLATFORM_OSX
+    FIREBASE_ASSERT_MESSAGE(
+        semaphore_ != SEM_FAILED,
+        "Failed to create semaphore. If running in sandbox mode be sure to "
```

Comment:
Maybe it would be good to include "Firebase" in the message?

Reply:
Done.